src/controllers/authenticationController.ts

```diff
@@ -200,6 +207,90 @@ export default class AuthenticationController {
       sessionTrackingId
     );
 
+    const errorOrMaybeAssertionRef = this.lollipopParams.isLollipopEnabled
+      ? await this.sessionStorage.getLollipopAssertionRefForUser(user)
+      : E.right(O.none);
+
+    if (E.isLeft(errorOrMaybeAssertionRef)) {
+      return ResponseErrorInternal(
+        "Error retrieving previous lollipop configuration"
+      );
+    }
+
+    if (
+      this.lollipopParams.isLollipopEnabled &&
+      O.isSome(errorOrMaybeAssertionRef.right)
+    ) {
+      // Sending a revoke message for previous assertionRef related to the same fiscalCode
+      // This operation is fire and forget
+      this.lollipopParams.lollipopService
+        .revokePreviousAssertionRef(errorOrMaybeAssertionRef.right.value)
+        .catch(err => {
+          log.error(
+            "acs: error sending revoke message for previous assertionRef [%s]",
+            err
+          );
+        });
+    }
+
+    // Delete the reference to CF and assertionRef for lollipop.
+    // This operation must be performed even if the lollipop FF is disabled
+    // to avoid inconsistency on CF-key relation if the FF will be re-enabled.
+    const errorOrDelLollipop = await this.sessionStorage.delLollipopAssertionRefForUser(
+      spidUser.fiscalNumber
+    );
+
+    if (E.isLeft(errorOrDelLollipop) || !errorOrDelLollipop.right) {
+      return ResponseErrorInternal("Error on LolliPoP initialization");
+    }
+
+    const maybeRequestId = getRequestIDFromResponse(
+      new DOMParser().parseFromString(
+        errorOrSpidUser.right.getAssertionXml(),
+        "text/xml"
+      )
+    );
+
+    if (
+      this.lollipopParams.isLollipopEnabled &&
+      O.isSome(maybeRequestId) &&
+      AssertionRef.is(maybeRequestId.value)
+    ) {
+      const assertionRef: AssertionRef = maybeRequestId.value;
+
+      const errorOrActivatedPubKey = await pipe(
+        AP.sequenceT(TE.ApplicativeSeq)(
+          pipe(
+            this.lollipopParams.lollipopService.activateLolliPoPKey(
+              assertionRef,
+              user.fiscal_code,
+              errorOrSpidUser.right.getAssertionXml()
+            )
+          ),
+          pipe(
+            TE.tryCatch(
+              () =>
+                this.sessionStorage.setLollipopAssertionRefForUser(
+                  user,
+                  assertionRef
+                ),
+              E.toError
+            ),
+            TE.chain(TE.fromEither),
```

Comment:
```suggestion
            TE.chainEitherKW,
```

Reply:
It should be `TE.chainEitherK(identity)`. What do you prefer?

Reply:
Fixed in [aed172f](https://github.com/pagopa/io-backend/pull/981/commits/aed172f75e19de6b4b268061997edf6703871fed)